Libraries/Opc.Ua.PubSub/Transport/MqttPubSubConnection.cs

```diff
@@ -670,23 +670,31 @@ private MqttClientOptions GetMqttClientOptions()
                     MqttTlsOptions mqttTlsOptions =
                         ((MqttClientProtocolConfiguration)transportProtocolConfiguration).MqttTlsOptions;
 
+                    List<X509Certificate2> x509Certificate2s = new List<X509Certificate2>();
+                    if (mqttTlsOptions?.Certificates != null)
+                    {
+                        foreach (var x509cert in mqttTlsOptions?.Certificates.X509Certificates)
+                        {
+                            x509Certificate2s.Add(new X509Certificate2(x509cert));
```

Comment:
Do these have to be disposed eventually?  Why not use the configured cert here and create a new one?

Reply:
They have to be disposed; I expected that the Mqtt client does it but looking into its code it does not.
The new call expects a X509Certificate2 type: WithClientCertificates(IEnumerable<X509Certificate2> certificates) therefore using the x509cert.Handle of the existing certificate should keep the same resource reference